src/docfx/build/moniker/MonikersProvider.cs

```diff
@@ -96,5 +96,17 @@ public List<string> GetZoneMonikers(string rangeString, List<string> fileLevelMo
             monikers.Sort(Comparer);
             return monikers;
         }
+
+        public string GetGroupIdFromMonikerList(IEnumerable<string> monikers)
+        {
+            if (monikers == null || monikers.Count() == 0)
+            {
+                return null;
+            }
+            var sortedMonikers = monikers.ToList();
+            sortedMonikers.Sort(Comparer);
```

Comment:
Seems there are many duplicates `Sort` here. Add some cache?

Reply:
yes, but this is a utility, we have to make sure the output is stable

---
In reply to: [240448390](https://github.com/dotnet/docfx/pull/3750#discussion_r240448390) [](ancestors = 240448390)